tst/SdpApiTest.cpp

```diff
@@ -228,6 +228,131 @@ TEST_F(SdpApiTest, setTransceiverPayloadTypes_HasRtxType) {
+    KvsPeerConnection->isOffer = TRUE;
+
+    RtcMediaStreamTrack track;
+    PRtcRtpTransceiver pTransceiver;
+    pTransceiver = (PRtcRtpTransceiver) MEMCALLOC(1, SIZEOF(RtcRtpTransceiver));
+
+    MEMSET(&track, 0x00, SIZEOF(RtcMediaStreamTrack));
+
+    track.kind = MEDIA_STREAM_TRACK_KIND_VIDEO;
+    track.codec = RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE;
+    STRCPY(track.streamId, "myKvsVideoStream");
+    STRCPY(track.trackId, "myVideoTrack");
+
+    pTransceiver->direction = RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY;
+    EXPECT_EQ(STATUS_SUCCESS, addTransceiver(offerPc, &track, NULL, &pTransceiver));
+
+    RtcSessionDescriptionInit sessionDescriptionInit;
+
+    EXPECT_EQ(STATUS_SUCCESS, hashTableCreate(&(KvsPeerConnection->pCodecTable)));
+    EXPECT_EQ(STATUS_SUCCESS, hashTablePut(KvsPeerConnection->pCodecTable, RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE, 1));
+    EXPECT_EQ(STATUS_SUCCESS, hashTableCreate(&(KvsPeerConnection->pRtxTable)));
+    EXPECT_EQ(STATUS_SUCCESS, hashTablePut(KvsPeerConnection->pRtxTable, RTC_RTX_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE, 2));
+
+    EXPECT_EQ(STATUS_SUCCESS, doubleListCreate(&(KvsPeerConnection->pTransceievers)));
+    EXPECT_EQ(STATUS_SUCCESS, doubleListInsertItemHead(KvsPeerConnection->pTransceievers, (UINT64) (pTransceiver)));
+    EXPECT_EQ(STATUS_SUCCESS, setTransceiverPayloadTypes(KvsPeerConnection->pCodecTable, KvsPeerConnection->pRtxTable, KvsPeerConnection->pTransceievers));
+
+    EXPECT_EQ(STATUS_SUCCESS, createOffer(offerPc, &sessionDescriptionInit));
+
+    EXPECT_PRED_FORMAT2(testing::IsSubstring, "sendonly", sessionDescriptionInit.sdp);
+    freePeerConnection((PRtcPeerConnection*) KvsPeerConnection);
+}
+
+TEST_F(SdpApiTest, populateSingleMediaSection_RecvOnlyDirection) {
+
+    PHashTable pCodecTable;
+    PHashTable pRtxTable;
+    PRtcPeerConnection offerPc = NULL;
+    RtcConfiguration configuration;
+    // Create peer connection
+    EXPECT_EQ(createPeerConnection(&configuration, &offerPc), STATUS_SUCCESS);
+    PKvsPeerConnection KvsPeerConnection = (PKvsPeerConnection) (offerPc);
+    KvsPeerConnection->isOffer = TRUE;
+
+    RtcMediaStreamTrack track;
+    PRtcRtpTransceiver pTransceiver;
+    pTransceiver = (PRtcRtpTransceiver) MEMCALLOC(1, SIZEOF(RtcRtpTransceiver));
+
+    MEMSET(&track, 0x00, SIZEOF(RtcMediaStreamTrack));
+
+    track.kind = MEDIA_STREAM_TRACK_KIND_VIDEO;
+    track.codec = RTC_CODEC_H264_PROFILE_42E01F_LEVEL_ASYMMETRY_ALLOWED_PACKETIZATION_MODE;
+    STRCPY(track.streamId, "myKvsVideoStream");
+    STRCPY(track.trackId, "myVideoTrack");
+
+    pTransceiver->direction = RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY;
+    EXPECT_EQ(STATUS_SUCCESS, addTransceiver(offerPc, &track, NULL, &pTransceiver));
+
+    RtcSessionDescriptionInit sessionDescriptionInit;
+
+    EXPECT_EQ(STATUS_SUCCESS, hashTableCreate(&(KvsPeerConnection->pCodecTable)));
```

Comment:
Do you need to call any of the private APIs for the test? I think you just need to add `addTransceiver` -> `createOffer`

Reply:
No. I had removed them in the local. Messed something up and the changes did not go in. Fixed it now.